`preserveSymlinks` breaks react types

While testing with vaul, if we don't set `preserveSymlinks` to `false`, it will break ts resolving

```
src/index.tsx(741,15): error TS2339: Property 'children' does not exist on type 'Omit<DialogOverlayPro
ps & RefAttributes<HTMLDivElement>, "ref">'.
src/index.tsx(748,9): error TS2322: Type '{ forceMount?: true; key?: Key; onMouseUp: (event: PointerEv
ent<HTMLDivElement>) => void; ref: (node: HTMLDivElement) => void; "vaul-drawer-visible": string; "vau
l-overlay": string; "vaul-snap-points": string; "vaul-snap-points-overlay": string; }' is not assignab
le to type 'IntrinsicAttributes & DialogOverlayProps & RefAttributes<HTMLDivElement>'.
  Property 'onMouseUp' does not exist on type 'IntrinsicAttributes & DialogOverlayProps & RefAttribute
s<HTMLDivElement>'.
```

We need to add a test case to cover this

--- a/src/build-config.ts
+++ b/src/build-config.ts
@@ -230,6 +230,9 @@ async function buildInputConfig(
         checkJs: false,
         declarationMap: false,
         skipLibCheck: true,
+        // preserveSymlinks should always be set to false to avoid issues with
+        // resolving types from <reference> from node_modules
+        preserveSymlinks: false,
         target: 'ESNext',
         ...(!tsCompilerOptions.jsx
           ? {

